canonical/traefik-k8s-operator
# Purge SDI from ingress (per app)

Removed SDI from ingress (per app)


## Context
See #37 


## Testing Instructions
- Added integration tests for 
  - ingress: spring-music
  - ingress-per-unit: prometheus-k8s

The relation interface is compatible with older versions of `ingress`, but the API is different. Similarly to ingress-per-unit, the `request/response` language was stripped in favour of a more semantically close one. See #42 and #48

the Requirer has `ready` and `revoked` events: reflecting the fact that the only thing that can happen to ingress is: it can be provided (with a given url), or it can be revoked (i.e. the url is no longer available).
`ready` also includes when ingress was already ready; i.e. when the url has changed.

the Provider has now 'data_provided` and `data_removed` events, following a similar principle: either the requirer has published the data we need to provide ingress, (or changed it), or it has wiped it.

Fixes #37 
Fixes #28 

## Release Notes
- Stripped SDI from ingress interface
- Refactored and simplified the ingress API
- Added itests that check that ingress is compatible with old SDI-based ingress (as implemented by spring-music)
- Added itests that check that ingress-per-unit is compatible with old SDI-based ingress-per-unit (as implemented by prometheus-k8s)
